src/v/cluster/client_quota_store.h

```diff
@@ -0,0 +1,75 @@
+// Copyright 2024 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+#pragma once
+
+#include "client_quota_serde.h"
+#include "container/fragmented_vector.h"
+
+#include <absl/algorithm/container.h>
+#include <absl/container/node_hash_map.h>
+
+namespace cluster::client_quota {
+
+class store final {
+public:
+    using container_type = absl::node_hash_map<entity_key, entity_value>;
```

Comment:
I think it needs to be `entity_key` because it could be a combination of user + client_id when it comes to discovering the rules (and once we support quotas based on principal)

Reply:
Yeah it has to be `entity_key` for the sake of compound keys, but we can use variant for the parts of the `entity_key`. I'm changing that now.

Reply:
Got it, I forget about compound keys.